src/drivers/intel/hda/hda-dma.c

```diff
@@ -309,9 +311,14 @@ static void hda_dma_post_copy(struct dma_chan_data *chan, int bytes)
 		 */
 		hda_dma_inc_fp(chan, bytes);
 
-		/* Force Host DMA to exit L1 if scheduled on DMA */
+		/*
+		 * Force Host DMA to exit L1 if scheduled on DMA,
+		 * otherwise, perform L1 exit at LL scheduler powt run.
+		 */
 		if (!hda_chan->irq_disabled)
 			pm_runtime_put(PM_RUNTIME_HOST_DMA_L1, 0);
```

Comment:
@keyonjie we do a pm_runtime_get() on copy without the check for irq_disabled. So I wondering if we still need the check to do the pm_runtime_put() here ?

Reply:
@ranj063 yes, we still need the check and do _put() in different ways for DMA IRQ driver and Timer driver scenarios, that is same with what @tlauda implemented. The only change is postpone the _get() for timer driver case.